.github/workflows/ci-pr-validation.yml

```diff
@@ -58,7 +58,6 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.nodejs }}
-          cache: 'npm'
```

Comment:
@tisonkun I didn't change to using yarn cache because using yarn cache requires a yarn.lock file. Referring to this https://github.com/apache/pulsar-client-node/pull/258#issuecomment-1342579551, we don't add yarn.lock.

For this compilation step, I chose not to use a cache.

Reply:
If we choose to check in the npm lock file, it's strange that we run CI tasks with `yarn`.

What if moving the pkg manager to `yarn` totally?

Reply:
> What if moving the pkg manager to yarn totally?

I think we'd better not. The motivation is to prevent the regression happened again. I think we can add a separated workflow to verify the `yarn install` in a single env like Linux.

Reply:
> I think we'd better not. The motivation is to prevent the regression happened again. I think we can add a separated workflow to verify the yarn install in a single env like Linux.

Reuse the current workflow to save CI execution time. `yarn install` command can take tens of seconds.

Reply:
The workflow is executed concurrently. And it would not take too much time. See the x64 build for Node 18 on Linux, it only takes 5 minutes in total.

Reply:
Ok, Changed.